packages/gatsby/src/schema/node-model.js

```diff
@@ -89,8 +106,18 @@ class LocalNodeModel {
   async runQuery(args, pageDependencies) {
     const { query, firstOnly, type } = args || {}
 
+    // We don't support querying union types (yet?), because the combined types
+    // need not have any fields in common.
```

Comment:
We can rethink not supporting union types in `runQuery` later, but it should be part of refactoring resolving nodes and run-sift (e.g. we cannot call `getFields()` on a union type there). Note that `getAllNodes` etc. *do* support both unions and interfaces.